client/checkout/platform-checkout/express-button/express-checkout-iframe.js

```diff
@@ -0,0 +1,226 @@
+					value: iframeHeaderValue,
+				},
+				getWooPayExpressData( 'platformCheckoutHost' )
+			);
+		}
+
+		// Prevent scrolling when the iframe is open.
+		document.body.style.overflow = 'hidden';
+	};
+
+	/**
+	 * Handles setting the iframe position based on the window size.
+	 * It tries to be positioned at the center of the screen unless
+	 * window is smaller than breakpoint which makes it full window size.
+	 */
+	const setPopoverPosition = () => {
+		// If for some reason the iframe is not loaded, just return.
+		if ( ! iframe ) {
+			return;
+		}
+
+		// If the window width is less than the breakpoint, set iframe to full window
+		if ( fullScreenModalBreakpoint >= window.innerWidth ) {
+			iframe.style.left = '0';
+			iframe.style.right = '';
+			iframe.style.top = '0';
+			return;
+		}
+
+		// Get references to the iframe bounding rects.
+		const iframeRect = iframe.getBoundingClientRect();
+
+		// Set the iframe top and left to be centered.
+		iframe.style.top =
+			Math.floor( window.innerHeight / 2 - iframeRect.height / 2 ) + 'px';
+		iframe.style.left =
+			Math.floor( window.innerWidth / 2 - iframeRect.width / 2 ) + 'px';
+	};
+
+	iframe.addEventListener( 'load', () => {
+		// Set the initial value.
+		iframeHeaderValue = true;
+
+		getWindowSize();
+		window.addEventListener( 'resize', getWindowSize );
+
+		setPopoverPosition();
+		window.addEventListener( 'resize', setPopoverPosition );
+
+		iframe.classList.add( 'open' );
+		wcpayTracks.recordUserEvent(
+			wcpayTracks.events.PLATFORM_CHECKOUT_OTP_START
+		);
+	} );
+
+	// Add the iframe to the wrapper.
+	iframeWrapper.insertBefore( iframe, null );
+
+	// Error message to display when there's an error contacting WooPay.
+	const errorMessage = document.createElement( 'div' );
+	errorMessage.style[ 'white-space' ] = 'normal';
+	errorMessage.textContent = __(
+		'WooPay is unavailable at this time. Please complete your checkout below. Sorry for the inconvenience.',
+		'woocommerce-payments'
+	);
+
+	const closeIframe = () => {
+		window.removeEventListener( 'resize', getWindowSize );
+		window.removeEventListener( 'resize', setPopoverPosition );
+
+		iframeWrapper.remove();
+		iframe.classList.remove( 'open' );
+
+		document.body.style.overflow = '';
+	};
+
+	iframeWrapper.addEventListener( 'click', closeIframe );
+
+	const openIframe = ( email ) => {
+		const urlParams = new URLSearchParams();
+		urlParams.append( 'email', email );
+		urlParams.append(
+			'needsHeader',
+			fullScreenModalBreakpoint > window.innerWidth
+		);
+		urlParams.append(
+			'wcpayVersion',
+			getWooPayExpressData( 'wcpayVersionNumber' )
+		);
+
+		iframe.src = `${ getWooPayExpressData(
+			'platformCheckoutHost'
+		) }/otp/?${ urlParams.toString() }`;
+
+		// Insert the wrapper into the DOM.
+		parentDiv.insertBefore( iframeWrapper, null );
+
+		setPopoverPosition();
+
+		// Focus the iframe.
+		iframe.focus();
+	};
+
+	const showErrorMessage = () => {
+		parentDiv.insertBefore( errorMessage );
```

Comment:
Definitely not as part of this PR but we should have a separate issue and consult with the design about the placement of the error message. Previously the OTP modal was triggered with the email field so it made sense to show the error message below the email field. But with the express checkout button which will even work with an empty email field, that place does not feel suitable anymore.

Reply:
Good call, this actually errors out for me in console. Created an issue to track this [here](https://github.com/Automattic/woopay/issues/1490).